dashboard/app/views/dynamic_config/dcdo_show.html.haml

```diff
@@ -0,0 +1,68 @@
+:scss
+  .dcdo-list {
+    .dcdo-padding {
+      padding: 10px;
+      margin: auto 0;
+    }
+
+    .dcdo-row {
+      display: grid;
+      grid-template-columns: 240px 384px 111px 100px;
+      border-bottom: 1px solid #eee;
+    }
+
+    .dcdo-submit-button {
+      margin: auto;
+    }
+
+    .dcdo-value {
+      input {
+        width: 350px;
+      }
+    }
+
+    select {
+      width: auto;
+    }
+
+    form, input, select {
+      margin: 0;
+    }
+  }
+
+- data_types = ["Array", "Boolean", "Float", "Hash", "Integer", "String"]
+
+.dcdo-list
+  %h3 DCDO Flags
+  %div
+    %div{:class => %w[dcdo-row]}
+      %div{:class => %w[dcdo-label dcdo-padding]} Key
+      %div{:class => %w[dcdo-label dcdo-padding]} Value
+      %div{:class => %w[dcdo-label dcdo-padding]} Data Type
+      %div{:class => %w[dcdo-label]}
+    = form_for :add, :url => {:controller => 'dynamic_config', :action => 'dcdo_set'} do |f|
```

Comment:
I'm not familiar with this usage of `form_for`; I expect the first argument to that method to be a Rails model object whose attributes the form is providing a means to update, and then for us to use the form value `f` within the method block to create the inputs. It doesn't look like we're doing either of those things here; should this be [`form_tag`](https://api.rubyonrails.org/v6.1.7.3/classes/ActionView/Helpers/FormTagHelper.html#method-i-form_tag) instead?

Reply:
Ah. Looks like Staring with Rails 5.1, [both are soft-deprecated](https://guides.rubyonrails.org/v6.1/form_helpers.html#using-form-tag-and-form-for) in favor of [`form_with`](https://api.rubyonrails.org/v6.1.7.6/classes/ActionView/Helpers/FormHelper.html#method-i-form_with). I'll update to use that.

Reply:
I switched to `form_with` but after hours poring over the docs for the `url` parameter, I could not figure out how they work and went with a hard coded URL path. I could use some guidance here.